### PR TITLE
rblake3sum: init at 0.4.0

### PR DESCRIPTION
**Pull request by dpc**
## Description of changes

[rblake3sum](https://github.com/rustshop/rblake3sum) is a recursive blake3 digest (hash) of a file-system path. It's a fork of [rblake2sum](https://github.com/crev-dev/rblake2sum) with a newer/better/faster hashing algorithm - both written and maintained by me.

While at it:
rblake2sum: set mainProgram

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by dpc** [comment]
> The package seems to have only one binary, rblake3sum; it could be set as meta's mainProgram.

Is this recommended? I always thought that `mainProgram` is useful when there's more than one binary only.

**Comment by onemoresuza** [comment]
> > The package seems to have only one binary, rblake3sum; it could be set as meta's mainProgram.
> 
> Is this recommended? I always thought that `mainProgram` is useful when there's more than one binary only.

I think the error message when running `lib.getExe pkgs.rblake3sum` may clear this up:

> trace: warning: getExe: Package "rblake3sum-0.4.0" does not have the meta.mainProgram attribute. We'll assume that the main program has the same name for now, but this behavior is deprecated, because it leads to surprising errors when the assumption does not hold. If the package has a main program, please set `meta.mainProgram` in its definition to make this warning go away. Otherwise, if the package does not have a main program, or if you don't control its definition, use getExe' to specify the name to the program, such as lib.getExe' foo "bar".

**Comment by dpc** [comment]
Good to know. While at it I've set it on rblake2sum. I hope it's OK.

**Comment by onemoresuza** [comment]
It is, but could add the `rblake2sum` change to the description?

**Review by onemoresuza — APPROVED** [review]

File: pkgs/tools/security/rblake3sum/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+  cargoHash = "sha256-SE/Zg/UEV/vhB/VDcn8Y70OUIoxbJBh6H2QgFMkWPc4=";
+
+  buildInputs = lib.optionals stdenv.isDarwin [ Security ];
+
+  meta = with lib; {
+    description = "A recursive blake3 digest (hash) of a file-system path";
+    homepage = "https://github.com/rustshop/rblake3sum";
+    license = [ licenses.mit ];
+    maintainers = with maintainers; [ dpc ];
```
[the remark this answers is not in the document]
Comment (dpc) [reply pkgs/tools/security/rblake3sum/default.nix:27]: Is it about the order? Alphabetical preferred?

[the remark this answers is not in the document]
Comment (onemoresuza) [reply pkgs/tools/security/rblake3sum/default.nix:26]: Do you mean this suggestion? If so, no: It was made before `meta.mainProgram` was added; it was just to add it.

**Review by pbsds — APPROVED** [review]
Result of `nixpkgs-review pr 259337` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>rblake3sum</li>
  </ul>
</details>

Seems to work as intended, LGTM. Thanks!

**Pull request merged**